### PR TITLE
add TreeNodeEx and related fixes

### PR DESCRIPTION
**Pull request by dpethes**
Mostly Treenode related stuff: 
* Added TreeNodeEx
* SetNextTreeNodeOpen was already fixed in one of your branches, just not merged
* fixed comment in TreeNodeFlags
* ColorTarget was broken, PopupBg shifted the value of half of the enums, so setting styles sometimes worked improperly

### CODE REVIEW

**Comment by mellinoe** [comment]
Hey, thanks for the continued contributions! I'll take a look at this later tonight, but it looks good on first glance. I think you've uncovered why some parts of my styling look wrong 😄

**Comment by dpethes** [comment]
I was doing some styling work and wondered what is happening as well :D

By the way, cimgui is already updated to 1.50 - do you plan to start working on it sometime soon? If not then I can port the changes over the next week, I'm doing a lot of imgui stuff at the moment so I have some time for that.

File: src/ImGui.NET/ImGui.cs [hunk 1]
```diff
@@ -1052,21 +1052,26 @@ public static void SetTooltip(string text)
...
         }
 
         public static bool TreeNode(string label)
         {
             return ImGuiNative.igTreeNode(label);
         }
 
+        public static bool TreeNodeEx(string label, TreeNodeFlags flags = 0)
+        {
+            return ImGuiNative.igTreeNodeEx(label);
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: `flags` should be passed in here.

**Comment by mellinoe** [comment]
Just one thing looks off here; other than that it's good. Thanks again!

> By the way, cimgui is already updated to 1.50 - do you plan to start working on it sometime soon? If not then I can port the changes over the next week, I'm doing a lot of imgui stuff at the moment so I have some time for that.

Awesome, I hadn't seen the new release yet. I may also have some time this weekend to make some updates.

**Comment by dpethes** [comment]
Thanks for review! I'll focus on the string handling next.

**Comment by mellinoe** [comment]
LGTM. Thanks!

**Pull request merged**